lib/import-export-visitor.js

```diff
@@ -140,13 +157,18 @@ module.exports = class ImportExportVisitor extends Visitor {
       this.bodyInfos.push(bodyInfo);
     }
 
+    bodyInfo.needToAddUseStrictDirective =
+      needToAddUseStrictDirective ||
+      bodyInfo.needToAddUseStrictDirective;
+
     return bodyInfo;
   }
 
   hoistImports(importDeclPath, hoistedCode, childName) {
     this.preserveChild(importDeclPath, childName);
     const bodyInfo = this.getBlockBodyInfo(importDeclPath);
     bodyInfo.hoistedImportsString += hoistedCode;
+    this.madeChanges = true;
     return this;
```

Comment:
I dig the "madeChanges" as a way to lazily detect if you've hit an `import` or `export`. I wonder if the name should be changed *(bikeshedding)* since it only relates to hoisted import/exports.